tools/release/release.mts

```diff
@@ -0,0 +1,67 @@
+import { execaSync } from 'execa';
+import {
+  releaseChangelog,
+  releasePublish,
+  releaseVersion,
+} from 'nx/src/command-line/release';
+import yargs from 'yargs';
+
+const options = await yargs
+  .version(false)
+  .option('version', {
+    description:
+      'Explicit version specifier to use, if overriding conventional commits',
+    type: 'string',
+  })
+  .option('dryRun', {
+    alias: 'd',
+    description:
+      'Whether to perform a dry-run of the release process, defaults to true',
+    type: 'boolean',
+    default: true,
+  })
+  .option('verbose', {
+    description: 'Whether or not to enable verbose logging, defaults to false',
+    type: 'boolean',
+    default: false,
+  })
+  .parseAsync();
+
+const { workspaceVersion, projectsVersionData } = await releaseVersion({
+  specifier: options.version,
+  // stage package.json updates to be committed later by the changelog command
+  stageChanges: true,
+  dryRun: options.dryRun,
+  verbose: options.verbose,
+});
+
+// Update the lock file after the version bumps and stage it ready to be committed by the changelog step
+if (!options.dryRun) {
+  console.log('⏳ Updating yarn.lock...');
+  execaSync(`yarn`, [`install`], {
+    env: { ...process.env, SKIP_POSTINSTALL: 'true' },
+  });
+  execaSync(`git`, [`add`, `yarn.lock`]);
+  console.log('✅ Updated and staged yarn.lock\n');
+}
+
+// This will create a release on GitHub
+await releaseChangelog({
+  versionData: projectsVersionData,
+  version: workspaceVersion,
+  dryRun: options.dryRun,
+  verbose: options.verbose,
+});
+
+// An explicit null value here means that no changes were detected across any package
+// eslint-disable-next-line eqeqeq
+if (workspaceVersion === null) {
+  console.log(
+    '⏭️ No changes detected across any package, skipping publish step altogether',
+  );
+} else {
+  await releasePublish({
+    dryRun: options.dryRun,
+    verbose: options.verbose,
+  });
+}
```

Comment:
I see you removed the process.exit - I believe we can’t do without it otherwise the Nx daemon will cause the process to not exit. You should be able to verify locally either way

Reply:
Interesting, wouldn't the file finishing execution be functionally the same as `process.exit(0)` and throwing an error be functionally the same as `process.exit(int)`? What's the difference from the Nx daemon's perspective?

Reply:
The internals of the utilities create a child process which communicates with the long running daemon server. This child process can cause it's parent to stay alive without an explicit exit.

Having said all that, I think I forgot about a later change to nx release utilities it does seem to no longer be required in the consuming script so that's good!